docs/docs/integrations/promotions/voucherify.md

```diff
@@ -0,0 +1,119 @@
+
+```mermaid
+sequenceDiagram
+  participant B as Browser
+  participant CV as @composable/voucherify
+  participant CGS as Commerce Generic
+  participant V as Voucherify API
+  B->>CGS: Discount related methods (addVoucherToCart, updateCartDiscount etc.)
+    activate CV
+  CGS->>CV: Use Voucherify specific methods from @composable/voucherify
+  activate CGS
+  CGS-->>CV: Cart object
+  deactivate CGS
+  Note over CV: Get cart vouchers codes from storage
+  CV->>V: Get available cart promotions
+  activate V
+  V-->>CV: promotions
+  deactivate V
+  CV->>V: Validate promotions and voucher codes
+  activate V
+  V-->>CV: Eligibility check and discount calculations results
+  deactivate V
+  Note over CV: Extend cart by discounts details
+  CV-->>B: Cart object containing discounts from Voucherify
+  deactivate CV
+```
+
+## Reference Files
+
+### Backend Files
+
+- `packages/voucherify`: This package contains functions that enhance the default behavior of functions related to discounts.
+
+### React Components
+
+- `composable-ui/src/components/cart/cart-vouchers.tsx`: A component that allows you to add, display, validate and delete vouchers.
+- `composable-ui/src/components/forms/voucher-form.tsx`: Form used in `cart-vouchers.tsx`.
+- `composable-ui/src/components/cart/cart-promotions.tsx`: Displays auto-applied cart promotions.
+
+## Integrating Voucherify with Composable UI
+
+### First steps
+
+1. [Create a Voucherify account](https://app.voucherify.io/#/signup).
+2. In Voucherify Dashboard, [set Discount Application Rule to "Partial"](https://support.voucherify.io/article/604-stacking-rules#application-rules)
+   Please go to:
+- `Redemptions` tab on the left side of the Voucherify dashboard
+- Select `Stacking rules`
+- In `Discounts' application rules` select `Edit settings`
+- Change the `Application rule` to `Partial - inapplicable (not met criteria) discounts are omitted`
+3. Retrieve your API keys from your Voucherify dashboard and set the following environment variables:
+
+```code
+VOUCHERIFY_API_URL=https://api.voucherify.io
+VOUCHERIFY_APPLICATION_ID=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxxx
+VOUCHERIFY_SECRET_KEY=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxxx
+```
+
+**Caution:** Ensure you never expose your Voucherify API keys in the NEXT_PUBLIC_* environment variables or client-side code. Take the necessary steps to ensure that secret keys are never disclosed to the public.
+
+
+### Populating the Products Using the Script
+
+To configure product base promotions in Voucherify, propagate product definitions to your Voucherify account:
+
+1. Open the terminal and navigate to the `scripts` directory.
+2. In the `scripts` directory, run the following command:
+  ```
+  pnpm install
+  ```
+3. To set up Voucherify, run the following command:
+  ```
+  pnpm voucherify-setup
```

Comment:
@marcin-slezak @weronika-kurczyna I think should this be `pnpm voucherify-preconfigure`? There is no script `voucherify-setup`

Reply:
You are right, the wrong name remained in the documentation. Fixed.